### DISCUSSION TITLE
中英文混合输出，输出会带英语单词

### DISCUSSION THREAD

**Discussion by tiwentichat** [zai-org/ChatGLM3#225] [Bad Case]
昨天正好截了张图，另一个出现长段英文的没截图，改天再出现就在这儿补充（或者另外单开一个讨论？
<img width="562" alt="屏幕截图 2023-11-08 013554" src="https://github.com/THUDM/ChatGLM3/assets/125454744/67397324-0387-4447-af6f-3951e0f4b065">

**Comment by zRzRzRzRzRzRzR** [comment]
这确实是一个问题

**Comment by zRzRzRzRzRzRzR** [comment]
<img width="472" alt="f8e0e4a4910c5072e4198fd9537655c" src="https://github.com/THUDM/ChatGLM3/assets/93239683/d6473dd5-ae2f-42c5-ada0-5d0d02ffd9b6">

使用fschat调用也能复现，mark

**Comment by leavegee** [comment]
怎么办呢?

**Comment by zRzRzRzRzRzRzR** [comment]
暂时没有办法，这是一个badcase，模型之后会改进

**Reply by leavegee** [reply]
模型什么时候更新解决这个严重的问题呢?

**Reply by zRzRzRzRzRzRzR** [reply]
内容已经上报，但是没有收到计划

**Comment by li-dongxu** [comment]
po主的例子至少是单个词输出英文, 我觉得我今天跑的这个更离谱
(属于是学混了)
![image](https://github.com/THUDM/ChatGLM3/assets/34854209/3fc3b94a-6e5a-4d49-b148-c2ca282fc048)

**Reply by zRzRzRzRzRzRzR** [reply]
真实，这个问题一直存在

**Comment by catmeowjiao** [comment]
我问他鲁迅是谁，他说鲁迅never是周树人的pen名

**Comment by gitker** [comment]
>>> n("李明在一个小岛上享受一个美好的   ")
假期
夜晚
时光
夏日
下午
春节
日子
vacation
早晨
夏天
夕阳
晚年

以上是一个例子
不知道是不是训练方式的原因，在全中文的提示词下，chatglm3的每次预测结果的下一个token的前20个候选里，有很大概率同时出现中文和英文，而且英文的权重还挺高，其他模型上没有发现类似问题。似乎模型对语境的理解不够？

**Comment by gitker** [comment]
import torch
def print_tokens(new_tokens):
    for k in new_tokens.tolist()[0]:
        print(tokenizer.decode([k]))

def next_char(s):
    inputs = tokenizer(s, return_tensors="pt").input_ids.to('cuda')
    out=model(inputs) 
    next_token_logits = out.logits[:, -1, :]
    probs = torch.nn.functional.softmax(next_token_logits, dim=-1)
    _,next_tokens = torch.topk(probs, 20)
    print_tokens(next_tokens)

next_char("李明在一个小岛上享受一个美好的 ")

**Comment by Zacgod** [comment]
这是我直接部署的情况，不只是中英文混用，还会出现繁体字。
![image](https://github.com/THUDM/ChatGLM3/assets/37444039/908b6c34-bb94-404f-a755-f2027b19be08)

用自己的数据微调后也会有这种情况（默认的微调参数）。
![image](https://github.com/THUDM/ChatGLM3/assets/37444039/d7b87f74-1d94-4803-997e-6d91acac4e88)

**Reply by 1014670860** [reply]
请问现在修复了吗

**Comment by PoisonousBromineChan** [comment]
<img width="1725" alt="abb3bf910c21114b3cfd606fe3071a2" src="https://github.com/THUDM/ChatGLM3/assets/64314088/8f5b2de8-d6dd-43bc-88bf-f7e2e5e7e828">
肯定没修复，这是刚测试的

**Comment by sisuad** [comment]
没有，update 20230105

**Comment by zRzRzRzRzRzRzR** [comment]
权重没有更新， 这代模型估计不会在修复这个问题

**Reply by dfengpo** [reply]
那怎么办啊，这么严重的问题，直接放弃使用，把大家都劝退了
